setup.py

```diff
@@ -71,23 +69,27 @@ def _get_all_gcc_commands(self):
             all_gcc_commands.extend(local_gccs)
         return all_gcc_commands
 
-
     def _find_suitable_gcc_gpp(self):
         # lists all gcc version in PATH
         all_gccs = self._get_all_gcc_commands()
         for gcc in all_gccs:
             matching_gpp = gcc.replace("cc", "++")
-            print(f'Found suitable gcc/g++ version {gcc} {matching_gpp}')
-            return gcc, matching_gpp
+            if shutil.which(gcc) and shutil.which(matching_gpp):
+                print(f'Found suitable gcc/g++ version {gcc} {matching_gpp}')
+                return gcc, matching_gpp
```

Comment:
```suggestion
            gcc_path = shutil.which(gcc)
            gpp_path = shutil.which(matching_gpp)
            if  gcc_path and gpp_path:
                print(f'Found suitable gcc/g++ version {gcc} {matching_gpp}')
                return gcc_path, gpp_path
```

Reply:
we could potentially use [walrus operator](https://realpython.com/python-walrus-operator/) to make the code even nicer, but may make some people unhappy:

```python
if (gcc_path :=  shutil.which(gcc)) and (gpp_path := shutil.which(matching_gpp)):
  print(f'Found suitable gcc/g++ version {gcc} {matching_gpp}')
  return gcc_path, gpp_path
```

more beautiful, no?